specification/compute/resource-manager/Microsoft.Compute/RecommenderRP/models.tsp

```diff
@@ -0,0 +1,119 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+
+namespace Microsoft.Compute;
+
+/**
+ * Contains additional properties of a diagnostic
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model DiagnosticProperties {
+  /**
+   * Describes what are the supported resource types for a diagnostic.
+   */
+  supportedResourceTypes?: string[];
+}
+
+/**
+ * SpotPlacementScores API Input.
+ */
+model SpotPlacementScoresInput {
+  /**
+   * The desired regions
+   */
+  desiredLocations?: string[];
+
+  /**
+   * The desired virtual machine SKU sizes.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/missing-x-ms-identifiers" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  desiredSizes?: ResourceSize[];
+
+  /**
+   * Desired instance count per region/zone based on the scope.
+   */
+  desiredCount?: int32;
+
+  /**
+   * Defines if the scope is zonal or regional.
+   */
+  availabilityZones?: boolean;
+}
+
+/**
+ * SpotPlacementRecommender API response.
+ */
+model ResourceSize {
+  /**
+   * The resource's CRP virtual machine SKU size.
+   */
+  sku?: string;
+}
+
+/**
+ * SpotPlacementScores API response.
+ */
+model SpotPlacementScoresResponse {
+  /**
+   * The desired regions
+   */
+  desiredLocations?: string[];
+
+  /**
+   * The desired virtual machine SKU sizes.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/missing-x-ms-identifiers" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  desiredSizes?: ResourceSize[];
+
+  /**
+   * Desired instance count per region/zone based on the scope.
+   */
+  desiredCount?: int32;
+
+  /**
+   * Defines if the scope is zonal or regional.
+   */
+  availabilityZones?: boolean;
+
+  /**
+   * A placement score indicating the likelihood of successfully allocating the specified Spot VM(s), as well as the expected lifetimes of the Spot VM(s) after allocation.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/missing-x-ms-identifiers" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
```

Comment:
Why is this suppression added over some properties and not others?

Reply:
Updated